credentials.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+)
+
+const defaultDotDir = ".carina"
+const defaultNonDotDir = "carina"
+const xdgDataHomeEnvVar = "XDG_DATA_HOME"
+
+// CarinaCredentialsBaseDir get the current base directory for carina credentials
+func CarinaCredentialsBaseDir() (string, error) {
+	if os.Getenv(CarinaHomeDirEnvVar) != "" {
+		return os.Getenv(CarinaHomeDirEnvVar), nil
+	}
+	if os.Getenv(CredentialsBaseDirEnvVar) != "" {
+		return os.Getenv(CredentialsBaseDirEnvVar), nil
+	}
+
+	// Support XDG
+	if os.Getenv(xdgDataHomeEnvVar) != "" {
+		return filepath.Join(os.Getenv(xdgDataHomeEnvVar), defaultNonDotDir), nil
+	}
```

Comment:
Hey as long as we're still supporting an XDG env var to make those folks happy, I definitely prefer this singular home directory setup.